pyterrier/pipelines.py

```diff
@@ -506,6 +563,48 @@ def _apply_round(measure, value):
         'total' : len(retr_systems),
         'desc' : 'pt.Experiment'
     }
+    
+    
+    common_pipe, execution_retr_systems = _identifyCommon(retr_systems)
+    precompute_time = 0
+    if precompute_shared and common_pipe is not None: 
+        print("Precomputing results of %d topics on shared pipeline component %s" % (len(topics), str(common_pipe)), file=sys.stderr)
+
+        tqdm_args_precompute = tqdm_args.copy()
+        tqdm_args_precompute['desc'] = "pt.Experiment precomputation"
+                
+        from timeit import default_timer as timer
+        starttime = timer()
+        if batch_size is not None:
+            import math
+            tqdm_args_precompute['unit'] = 'batches'
+            # round number of batches up for each system
+            tqdm_args_precompute['total'] = math.ceil((len(topics) / batch_size))
+            with pt.tqdm(**tqdm_args_precompute) as pbar:
+                precompute_results = []
+                for r in common_pipe.transform_gen(topics, batch_size=batch_size):
+                    precompute_results.append(r)
+                    pbar.update(1)
+                execution_topics = pd.concat(precompute_results)
+        else:  
+            tqdm_args_precompute['total'] = 1 
+            with pt.tqdm(**tqdm_args_precompute) as pbar:
```

Comment:
Is this tqdm meant to show that the common prefix is running?

Reply:
yes